static/js/components/LeaveProgramWizard.js

```diff
@@ -0,0 +1,171 @@
+// @flow
+/* global SETTINGS: false */
+import React from "react"
+import type { Dispatch } from "redux"
+import { Card } from "react-mdl/lib/Card"
+import Dialog from "material-ui/Dialog"
+import SelectField from "material-ui/SelectField"
+import MenuItem from "material-ui/MenuItem"
+import R from "ramda"
+import { dialogActions } from "./inputs/util"
+import _ from "lodash"
+
+import ProfileFormFields from "../util/ProfileFormFields"
+import { showDialog, hideDialog, setProgramsToUnEnroll } from "../actions/ui"
+import {
+  unEnrollProgramEnrollments,
+  UNENROLL_PROGRAM_DIALOG
+} from "../actions/programs"
+import type { Profile } from "../flow/profileTypes"
+import type {
+  AvailableProgramsState,
+  AvailableProgram
+} from "../flow/enrollmentTypes"
+import type { UIState } from "../reducers/ui"
+
+const selectStyle = {
+  backgroundColor: "#ffff",
+  border:          "1px solid #c3c1c1",
+  padding:         "0 10px",
+  borderRadius:    "3px",
+  height:          "40px",
+  marginTop:       "7px"
+}
+const underlineStyle = {
+  borderBottom: "0",
+  bottom:       "0"
+}
+const hintStyle = {
+  color:  "#a2a2a2",
+  bottom: "5px"
+}
+const isVisible = R.propOr(false, UNENROLL_PROGRAM_DIALOG)
+
+class LeaveProgramWizard extends ProfileFormFields {
+  props: {
+    dispatch: Dispatch,
+    profile: Profile,
+    programs: AvailableProgramsState,
+    ui: UIState
+  }
+
+  setDialogVisibility = (visible: boolean) =>
+    visible
+      ? showDialog(UNENROLL_PROGRAM_DIALOG)
+      : hideDialog(UNENROLL_PROGRAM_DIALOG)
+
+  enrolledInPrograms = () => {
+    const { programs } = this.props
+    return R.filter((program: AvailableProgram) => program.enrolled, programs)
```

Comment:
I think this would be better as a partial `const` outside the class and then call that in `renderUnenrollUI()`